include/cpp-statsd-client/StatsdClient.hpp

```diff
@@ -71,15 +75,27 @@ class StatsdClient {
 
     //! The UDP sender to be used for actual sending
     std::unique_ptr<UDPSender> m_sender;
+
+    //! The random number generator for handling sampling
+    static thread_local std::mt19937 m_random_engine;
```

Comment:
its expensive to make these over and over, we could have one per object and that would be fine but since its thread local static can work in this case. let me know if youd like me to just make it a plain instance member variable though